### PR TITLE
Fix/opensea api update

### PR DESCRIPTION
**Pull request by trkaplan**
Opensea changed the API and started to require an API-key to make calls.
~~3rd party component is still legacy. This PR includes the patch for~~ `react-nft-gallery` : Component released a new version, so deleted the Patch and upgraded it to the latest version.
- Supports custom API endpoint (to inject API key on the Server side)
- Support new pagination mechanism 
- Added a Retry button to be shown on API error
- Fixed the typo
- Fixed the annoying layout resize on preloading.
- Chore: removed package-lock.json, updated yarn.lock since yarn is preferred for install in README.

Custom API repo: 
https://github.com/trkaplan/opensea-api-proxy

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,17 +7,19 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
+    "patch-package": "^6.4.7",
```
Comment (mdnorman) [inline package.json]: i did not know about this library. Huge improvement to some of the things I've had to do in the past.

File: patches/react-nft-gallery+0.6.2.patch [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
...0,0.1),0 10px 10px -5px rgba(0,0,0,0.04);box-shadow:var(--tw-ring-offset-shadow,0 0 transparent),var(--tw-ring-shadow,0 0 transparent),var(--tw-shadow)}*,:after,:before{--tw-ring-inset:var(--tw-empty,/*!*/ /*!*/);--tw-ring-offset-width:0px;--tw-ring-offset-color:#fff;--tw-ring-color:rgba(59,130,246,0.5);--tw-ring-offset-shadow:0 0 transparent;--tw-ring-shadow:0 0 transparent}.rnftg-transition{transition-property:background-color,border-color,color,fill,stroke,opacity,box-shadow,transform,filter,-webkit-backdrop-filter;transition-property:background-color,border-color,color,fill,stroke,opacity,box-shadow,transform,filter,backdrop-filter;transition-property:background-color,border-color,color,fill,stroke,opacity,box-shadow,transform,filter,backdrop-filter,-webkit-backdrop-filter;transition-timing-function:cubic-bezier(.4,0,.2,1);transition-duration:.15s}.rnftg-duration-300{transition-duration:.3s}@media (min-width:768px){.md\\:rnftg-grid-cols-2{grid-template-columns:repeat(2,minmax(0,1fr))}}@media (min-width:1024px){.lg\\:rnftg-grid-cols-3{grid-template-columns:repeat(3,minmax(0,1fr))}}@media (min-width:1280px){.xl\\:rnftg-grid-cols-4{grid-template-columns:repeat(4,minmax(0,1fr))}}";
```
Comment (mdnorman) [inline patches/react-nft-gallery+0.6.2.patch]: this is a bit opaque to have to add in this way. may want to check in the source that generated this

File: patches/react-nft-gallery+0.6.2.patch [hunk 3]
```diff
@@ -0,0 +1,422 @@
+diff --git a/node_modules/react-nft-gallery/dist/react-nft-gallery.esm.js b/node_modules/react-nft-gallery/dist/react-nft-gallery.esm.js
```
Comment (mdnorman) [inline patches/react-nft-gallery+0.6.2.patch]: this is a pretty big diff to have to do against the transpiled code. Understand the urgency, so makes sense to do it this way temporarily.

**Review by mdnorman — APPROVED** [review]
These fixes look good to me.

File: package.json (hunk 1 above)
Comment (trkaplan) [reply package.json]: I like it, life-saver :)

File: patches/react-nft-gallery+0.6.2.patch (hunk 2 above)
Comment (trkaplan) [reply patches/react-nft-gallery+0.6.2.patch]: You can see the diff here: https://github.com/trkaplan/react-nft-gallery/compare/main...trkaplan:custom-api-support?expand=1 
(Commits on April 3,4 is not included in this patch)

I sent a PR to to the 3rd party component's owner. He is about to be merge the changes to the master. After that I'll just upgrade the dependency and remove the patch.

File: patches/react-nft-gallery+0.6.2.patch (hunk 3 above)
Comment (trkaplan) [reply patches/react-nft-gallery+0.6.2.patch]: Right, see my comment above.

**Pull request merged**